pkg/analysis/signature.go

```diff
@@ -50,20 +57,24 @@ func (a *Analyzer) SignatureHelp(doc document.Document, pos protocol.Position) *
 	}
 }
 
+type callArguments struct {
+	positional, total uint32
+	keywords          []string
```

Comment:
IIUC, you could use a `map[string]bool` here to eliminate the risk / uncertainty about these not being stored in sorted order (since we're currently binary searching this)

(also makes clear that the order is thrown out, and the reader might not think they're in text order)

Reply:
Yeah, I think that might be a better fit here. Thanks!